From 0ef15567a5c4f78cb065f54bbf8a73d77ce353bc Mon Sep 17 00:00:00 2001
From: torvic9 <torvic9@mailbox.org>
Date: Tue, 10 Nov 2020 19:48:44 +0100
Subject: [PATCH] clang-ias-dwarf-fixes

(by Sedat Dilek)

Signed-off-by: torvic9 <torvic9@mailbox.org>
---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index 008aba5f1..3effea6af 100644
--- a/Makefile
+++ b/Makefile
@@ -827,12 +827,17 @@ DEBUG_CFLAGS	+= -gsplit-dwarf
 else
 DEBUG_CFLAGS	+= -g
 endif
 
+ifndef LLVM_IAS
 KBUILD_AFLAGS	+= -Wa,-gdwarf-2
+endif
 
 ifdef CONFIG_DEBUG_INFO_DWARF4
 DEBUG_CFLAGS	+= -gdwarf-4
+ifdef LLVM_IAS
+KBUILD_AFLAGS  += -Wa,-gdwarf-4
+endif
 endif
 
 ifdef CONFIG_DEBUG_INFO_REDUCED
 DEBUG_CFLAGS	+= $(call cc-option, -femit-struct-debug-baseonly) \
-- 
2.29.2

